src/AppInstallerCLICore/Commands/UninstallCommand.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "UninstallCommand.h"
+#include "Workflows/UninstallFlow.h"
+#include "Workflows/InstallFlow.h"
+#include "Workflows/CompletionFlow.h"
+#include "Workflows/WorkflowBase.h"
+#include "Resources.h"
+
+using AppInstaller::CLI::Execution::Args;
+using AppInstaller::CLI::Workflow::ExecutionStage;
+
+namespace AppInstaller::CLI
+{
+    std::vector<Argument> UninstallCommand::GetArguments() const
+    {
+        // TODO: determine exact arguments needed
+        return
+        {
+            Argument::ForType(Args::Type::Query),
+            Argument::ForType(Args::Type::Manifest),
+            Argument::ForType(Args::Type::Id),
+            Argument::ForType(Args::Type::Name),
+            Argument::ForType(Args::Type::Moniker),
+            Argument::ForType(Args::Type::Version),
+            Argument::ForType(Args::Type::Channel),
+            Argument::ForType(Args::Type::Source),
+            Argument::ForType(Args::Type::Exact),
+            Argument::ForType(Args::Type::Interactive),
+            Argument::ForType(Args::Type::Silent),
+            Argument::ForType(Args::Type::Log),
+        };
+    }
+
+    Resource::LocString UninstallCommand::ShortDescription() const
+    {
+        return { Resource::String::UninstallCommandShortDescription };
+    }
+
+    Resource::LocString UninstallCommand::LongDescription() const
+    {
+        return { Resource::String::UninstallCommandLongDescription };
+    }
+
+    void UninstallCommand::Complete(Execution::Context& context, Execution::Args::Type valueType) const
```

Comment:
>Complete [](start = 27, length = 8)

I expect that this is a copy of the upgrade command Complete function.  Is that correct?

Reply:
Yes, I copied it from there but then I removed some arguments that are not needed, like InstallLocation.